python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -399,9 +570,10 @@ def asset_partitions_def_for_output(self, output_name: str = "result") -> Partit
 
         return result
 
+    @deprecated
```

Comment:
really wish this was named `deprecated_in_docs` or something because I had to check that this didn't emit runtime warnings.